src/zenml/integrations/mlflow/model_deployers/mlflow_model_deployer.py

```diff
@@ -0,0 +1,311 @@
+                f"service was found."
+            )
+
+        # set the uuid in the config
+        config.caller_uuid = self.uuid
+        # create a new service for the new model
+        service = MLFlowDeploymentService(config)
+        service.start(timeout=config.timeout)
+
+        return service
+
+        
+    def find_model_server(
+        self,
+        pipeline_name: Optional[str] = None,
+        run_id: Optional[str] = None,
+        step_name: Optional[str] = None,
+        model_name: Optional[str] = None,
+        model_uri: Optional[str] = None,
+        model_type: Optional[str] = None,
+    ) -> List[MLflowDeploymentService]:
+        """Method to find one or more model servers that match the
+        given criteria.
+
+        Args:
+            pipeline_name: Name of the pipeline that the deployed model was part
+            of.
+            run_id: ID of the pipeline run which the deployed model was part of.
+            step_name: The name of the pipeline model deployment step that
+                deployed the model.
+            model_name: Name of the deployed model.
+            model_uri: URI of the deployed model.
+            model_type: Type/format of the deployed model. Not used in this MLflow
+                case.
+
+        Returns:
+            One or more Service objects representing model servers that match
+            the input search criteria.
+        """
+
+        services = []
+        config = MLflowDeploymentConfig(
+            model_name=model_name,
+            model_uri=model_uri,
+            pipeline_name=pipeline_name,
+            run_id=run_id,
+            step_name=step_name
+        )
+        services_path = os.path.join(
+            get_global_config_directory(),
+            LOCAL_STORES_DIRECTORY_NAME,
+            str(self.uuid),
+        )
+        
+        for root, dirs, files in os.walk(services_path):
+            for file in files:
+                if(file == SERVICE_DAEMON_CONFIG_FILE_NAME):
+                    service_config_path = os.path.join(root, file)                    
+                    logger.info(
+                        "Loading service daemon configuration from %s", service_config_path
+                    )
+                    old_config = None
+                    with open(service_config_path, "r") as f:
+                        old_config = f.read()
+                    old_service = ServiceRegistry().load_service_from_json(old_config)
+                    if not isinstance(old_service, MLflowDeploymentService):
+                        raise TypeError(
+                            f"Expected service type MLflowDeploymentService but got "
+                            f"{type(old_service)} instead"
+                        )
+                    if(self._matches_search_criteria(old_service, config)):
+                        services.append(old_service)
+        
+        return services
```

Comment:
Shouldn't you filter out dead services here ?

Reply:
Filtering out dead services here would mean that I will not be able to use this function inside `start_model_server` since all services that are returned will always be in a running state.